Implemented vestri-btn class

dusanradovanovic opened the pull request:
Description:
Implemented `vestri-btn` class. It should replace various ad-hoc classes used for buttons across pages.

Linting:
<!--Have you validated that no linting errors are introduced? -->
- [x] No linting errors

Tests:
<!--Have tests been run locally and passed? If manual tests run, explain what was run below-->
- [ ] Smoke tests (mvn clean install -Dcucumber.options="--tags @smoketest" from test)
- [x] Manual tests

Documentation:
<!--Are documentation updates required? Include any mention of updates required below if necessary-->
- [ ] Requires documentation updates

dusanradovanovic commented:
Take a look at the product page for a showcase of all possible button variations.

dusanradovanovic commented:
Tests are failing but apparently the are failing on master at the moment.

shaunmaharaj commented on src/components/addressform.main.jsx:
```diff
@@ -426,10 +426,10 @@ class AddressFormMain extends React.Component {
             <div className="control-label" />
             <div className="form-input btn-container">
-              <button className="btn btn-primary address-save-btn" data-el-label="addressForm.save" type="submit">
+              <button className="vestri-btn primary address-save-btn" data-el-label="addressForm.save" type="submit">
```
minor comment. Can we change the class name of the button from "vestri-btn" to "styled-btn" or something more generic? Just so when it comes to branding, the styles don't contain the "vestri" name.

dusanradovanovic replied:
Renamed to `ep-btn`

shaunmaharaj commented on src/containers/CheckoutPage.jsx:
```diff
@@ -426,7 +426,7 @@ class CheckoutPage extends React.Component {
               {deletable && (
                 <div className="payment-btn-cell">
-                  <button className="btn checkout-delete-payment-btn" type="button" onClick={() => { this.handleDelete(payment.self.uri); }}>
+                  <button className="vestri-btn small" type="button" onClick={() => { this.handleDelete(payment.self.uri); }}>
```
The independent class names (example: checkout-delete-payment-btn) are meant more for readability and for automation. I'd prefer to keep some or add IDs where applicable rather than having generic elements in some places.

shaunmaharaj commented on src/containers/CheckoutAuthPage.jsx:
```diff
@@ -141,7 +141,7 @@ class CheckoutAuthPage extends React.Component {
                     {intl.get('create-an-account-message')}
                   </p>
-                  <button className="btn btn-primary checkout-auth-option-register-btn" type="button" onClick={() => { history.push('/registration', { returnPage: '/checkout' }); }}>
+                  <button className="vestri-btn primary wide" type="button" onClick={() => { history.push('/registration', { returnPage: '/checkout' }); }}>
```
The independent class names (example: checkout-auth-option-register-btn) are meant more for readability and for automation. I'd prefer to keep some or add IDs where applicable rather than having generic elements in some places.

shaunmaharaj commented on src/components/profilepaymentmethods.main.jsx:
```diff
@@ -74,7 +74,7 @@ class ProfilePaymentMethodsMain extends React.Component {
                   </span>
                 </div>
-                <button className="btn profile-delete-payment-btn" type="button" onClick={() => { this.handleDelete(paymentElement.self.uri); }}>
+                <button className="vestri-btn small" type="button" onClick={() => { this.handleDelete(paymentElement.self.uri); }}>
```
The independent class names (example: profile-delete-payment-btn) are meant more for readability and for automation. I'd prefer to keep some or add IDs where applicable rather than having generic elements in some places.

shaunmaharaj commented on src/components/profileInfo.main.jsx:
```diff
@@ -171,10 +171,10 @@ class ProfileInfoMain extends React.Component {
                   {intl.get('save')}
                 </button>
-                <button className="btn address-cancel-btn profile-info-cancel-btn" data-el-label="addressForm.cancel" type="button" onClick={() => { this.cancel(); }}>
+                <button className="vestri-btn small" data-el-label="addressForm.cancel" type="button" onClick={() => { this.cancel(); }}>
```
Need unique ID or class name

shaunmaharaj commented on src/components/profileInfo.main.jsx:
```diff
@@ -171,10 +171,10 @@ class ProfileInfoMain extends React.Component {
               </div>
               <div className="form-group create-address-btn-container profile-info-btn-container">
-                <button className="btn address-save-btn profile-info-save-btn" data-el-label="addressForm.save" type="submit">
+                <button className="vestri-btn primary small profile-info-save-btn" data-el-label="addressForm.save" type="submit">
```
Need unique ID or class name

shaunmaharaj commented on src/components/profileaddresses.main.jsx:
```diff
@@ -101,10 +101,10 @@ class ProfileAddressesMain extends React.Component {
                   </ul>
                 </div>
-                <button className="btn edit-address-btn" type="button" onClick={() => { this.editAddress(addressElement.self.uri); }}>
+                <button className="vestri-btn small" type="button" onClick={() => { this.editAddress(addressElement.self.uri); }}>
```
Need unique ID or class name

shaunmaharaj commented on src/components/profileaddresses.main.jsx:
```diff
@@ -101,10 +101,10 @@ class ProfileAddressesMain extends React.Component {
                   {intl.get('edit')}
                 </button>
-                <button className="btn delete-address-btn" type="button" onClick={() => { this.handleDelete(addressElement.self.uri); }} data-actionlink="">
+                <button className="vestri-btn small" type="button" onClick={() => { this.handleDelete(addressElement.self.uri); }} data-actionlink="">
```
Need unique ID or class name

shaunmaharaj commented on src/components/paymentform.main.jsx:
```diff
@@ -331,10 +331,10 @@ class PaymentFormMain extends React.Component {
             <div className="control-label" />
             <div className="form-input btn-container">
-              <button className="btn btn-primary payment-save-btn" data-el-label="paymentForm.save" type="submit">
+              <button className="vestri-btn primary" data-el-label="paymentForm.save" type="submit">
```
Need unique ID or class name

shaunmaharaj commented on src/components/paymentform.main.jsx:
```diff
@@ -331,10 +331,10 @@ class PaymentFormMain extends React.Component {
                 {intl.get('continue')}
               </button>
-              <button className="btn payment-cancel-btn" data-el-label="paymentForm.cancel" type="button" onClick={() => { this.cancel(); }}>
+              <button className="vestri-btn" data-el-label="paymentForm.cancel" type="button" onClick={() => { this.cancel(); }}>
```
Need unique ID or class name

shaunmaharaj commented:
@dusanradovanovic No further comments for this one. If it passes smoke tests, then we can merge. Great job with this!!

dusanradovanovic commented:
I've renamed `vestri-btn` to `ep-btn`, returned descriptive classes to all buttons and removed "demo" of all buttons from the product page. So this is ready for merge now.
